src/rpc/mempool.cpp

```diff
@@ -812,6 +813,47 @@ static RPCHelpMan savemempool()
     };
 }
 
+static RPCHelpMan getorphantxs()
+{
+    return RPCHelpMan{"getorphantxs",
+        "\nShows transactions from the tx orphanage.\n"
+        "\nEXPERIMENTAL warning: this call may be changed in future releases.\n",
+        {},
+        {
+            RPCResult{"orphans",
+                RPCResult::Type::ARR, "", "",
+                {
+                    {RPCResult::Type::OBJ, "", "",
+                    {
+                        {RPCResult::Type::STR_HEX, "txid", "The transaction hash in hex"},
+                        {RPCResult::Type::STR_HEX, "wtxid", "The transaction witness hash in hex"},
+                    }},
+                }
+            },
+        },
+        RPCExamples{
+            HelpExampleCli("getorphantxs", "")
+            + HelpExampleRpc("getorphantxs", "")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    const NodeContext& node = EnsureAnyNodeContext(request.context);
+    PeerManager& peerman = EnsurePeerman(node);
+    std::vector<CTransactionRef> orphanage = peerman.GetOrphans();
+
+    UniValue ret(UniValue::VARR);
+    for (auto const& orphan : orphanage) {
+        UniValue o(UniValue::VOBJ);
+        o.pushKV("txid", orphan->GetHash().ToString());
```

Comment:
I feel like the ability to fetch the whole transaction (set) optionally is useful, otherwise we just have a couple hashes of the transactions, no ability to estimate how many vbytes the orphanage is using, do any analysis on the data given?

Maybe have an optional argument via txid/wtxid that fetches the full serialized transaction in hex?

Reply:
Agreed, will refinea

Reply:
Follow-up PRs can probably add more interesting fields as well, such as entry time, announcer peerid, etc.

I'll try not to scope creep this PR too bad

Reply:
> ability to estimate how many vbytes the orphanage is using

A `getorphanageinfo` can be added for overall stats, perhaps after #28031 which adds size tracking.